WooCommerce/WooCommerceTests/ViewRelated/Search/Order/OrderSearchUICommandTests.swift

```diff
@@ -0,0 +1,171 @@
+    }
+
+    func testCreateCellViewModel() {
+        // Given
+        let mockOrder = MockOrders().makeOrder(status: .onHold, items: [], shippingLines: [], refunds: [], fees: [], taxes: [], customFields: [], giftCards: [])
+
+        let command = OrderSearchUICommand(siteID: siteID, onSelectSearchResult: { _, _ in }, storageManager: storageManager)
+
+        // Insert mock order statuses
+        insertOrderStatuses()
+
+        // When
+        let cellViewModel = command.createCellViewModel(model: mockOrder)
+
+        // Then
+        XCTAssertNotNil(cellViewModel, "Expected createCellViewModel to return an OrderListCellViewModel instance")
+        XCTAssertEqual(cellViewModel.statusString, "on-hold", "Expected order status name to match")
+        XCTAssertEqual(cellViewModel.status, .onHold, "Expected createCellViewModel to return on hold status")
+    }
+
+    func testSanitizeKeyword() {
+        // Given
+        let command = OrderSearchUICommand(siteID: siteID, onSelectSearchResult: { _, _ in }, storageManager: storageManager)
+
+        // When
+        let sanitizedKeywordWithHash = command.sanitizeKeyword("#123")
+        let sanitizedKeywordWithoutHash = command.sanitizeKeyword("123")
+
+        // Then
+        XCTAssertEqual(sanitizedKeywordWithHash, "123", "Expected sanitizeKeyword to remove the leading '#'")
+        XCTAssertEqual(sanitizedKeywordWithoutHash, "123", "Expected sanitizeKeyword to return the keyword unchanged if there's no leading '#'")
+    }
+
+    func testSynchronizeModelsTracksAnalytics() throws {
+        // Given
+        let stores = MockStoresManager(sessionManager: .testingInstance)
+        let command = OrderSearchUICommand(siteID: siteID, onSelectSearchResult: { _, _ in }, storageManager: storageManager, analytics: analytics, stores: stores)
+        let keyword = "testKeyword"
+        stores.whenReceivingAction(ofType: OrderAction.self) { (action: OrderAction) in
+            guard case let .searchOrders(_, _, _, _, onCompletion) = action else {
+                return XCTFail("Unexpected action: \(action)")
+            }
+            onCompletion(nil)
+        }
+
+        // Mock order statuses
+        insertOrderStatuses()
+
+        // When
+        waitFor { promise in
+            command.synchronizeModels(siteID: self.siteID, keyword: keyword, pageNumber: 1, pageSize: 20) { success in
+                promise(())
+            }
+        }
+
+        // Then
+        guard let event = analyticsProvider.receivedEvents.first else {
+            XCTFail("Expected an event but found none")
+            return
+        }
+        XCTAssertEqual(event, "orders_list_search")
+
+        guard let eventProperties = analyticsProvider.receivedProperties.first else {
+            XCTFail("Expected event properties but found none")
+            return
+        }
+        XCTAssertEqual(eventProperties["search"] as? String, keyword)
```

Comment:
Analytic event testing is so gross in this whole setup... there's nothing wrong with any of what you've done here, just empathising that it's so wordy. We should improve that...

Reply:
I agree. Would be great to reduce the verbosity.